src/storage/compression/bitpacking.cpp

```diff
@@ -949,16 +966,37 @@ void BitpackingSkip(ColumnSegment &segment, ColumnScanState &state, idx_t skip_c
 	scan_state.Skip(segment, skip_count);
 }
 
+template <class T>
+unique_ptr<CompressedSegmentState> BitpackingInitSegment(ColumnSegment &segment, block_id_t block_id,
+                                                         optional_ptr<ColumnSegmentState> segment_state) {
+	// segment_state is not used for Bitpacking (no specialized serialize/deserialize methods)
+	// block_id would equal INVALID_BLOCK for a constant value block (not materialized)
+	// auto &buffer_manager = BufferManager::GetBufferManager(segment.db);
+	if (block_id == INVALID_BLOCK) {
+		return nullptr; // auto handle = buffer_manager.Pin(segment.block);
+	}
+	auto result = make_uniq<BitpackingSegmentState>();
+	// Use scan_state to trigger decoding of the (first group's?) metadata
```

Comment:
This might be correct? I have to admit im not sure when init segment is called with a persistent block

I do wonder how useful this information is though
I think it would be more useful to indicate how many blocks for each compression mode exist

I think you'll want to make of the Serialize and Deserialize callbacks and set this information on the segment state at a later point (when flushing the segment)

Reply:
You can have a look at StringUncompressed (might be the other  way around) for an example of serialize/deserialize

Reply:
I did look there, but understood this as callback for storing additional info that is not handled by the base. The Bitpacking mode is already in the core.

Reply:
Serialize/deserialize is used when the information cannot or can't cheaply be retrieved from just the segment

When we want to store the count of groups that use the compression modes we would need to scan the entire segment in the InitSegment, this currently scans the header+metadata for the first group to get the `state.current_group.mode` data

I imagine we want to serialize and deserialize this data instead

Reply:
If I would create such a "block histogram", what would be the natural place to return it?
Is there an existing function to reveal the total number of blocks for a segment, for example, that I should extend?

Reply:
No, I think you just stringify it like you've done with your initial implementation
It's not a great way to represent it but the information will be more useful in my opinion

Reply:
OK thanks that makes sense.

Any ideas how to resolve that the CHECKPOINT (SQL) is not triggering init_segment? 

(Maybe I can work around it as it does trigger serialize?)

Reply:
Are you sure it's not triggering `init_segment`?
I believe it *is* but you're explicitly returning nullptr for that, as I have explained

Please check if that's the case

Reply:
I'll look into finishing it soon, thanks for the help!

Reply:
So I just tried your suggested solution, and that works after CHECKPOINT (as the test shows), but it does not report the segment info when I reload the database - so we need a mix of the two solutions I suppose. I will look into it and get back to you here, A.

Reply:
Continued at #15622 (sorry, did something wrong while switching computers)